src/cli.rs

```diff
@@ -66,13 +67,40 @@ pub struct Rover {
     #[structopt(long = "output", default_value = "plain", possible_values = &["json", "plain"], case_insensitive = true, global = true)]
     output_type: OutputType,
 
+    /// Accept invalid certificates when performing HTTPS requests.
+    ///
+    /// You should think very carefully before using this flag.
+    ///
+    /// If invalid certificates are trusted, any certificate for any site will be trusted for use.
+    /// This includes expired certificates.
+    /// This introduces significant vulnerabilities, and should only be used as a last resort.
+    #[structopt(
+        long = "insecure-accept-invalid-certs",
+        case_insensitive = true,
+        global = true
+    )]
+    accept_invalid_certs: bool,
+
+    /// Accept invalid hostnames when performing HTTPS requests.
+    ///
+    /// You should think very carefully before using this flag.
+    ///
+    /// If hostname verification is not used, any valid certificate for any site will be trusted for use from any other.
+    /// This introduces a significant vulnerability to man-in-the-middle attacks.
+    #[structopt(
+        long = "insecure-accept-invalid-hostnames",
+        case_insensitive = true,
+        global = true
+    )]
+    accept_invalid_hostnames: bool,
+
     #[structopt(skip)]
     #[serde(skip_serializing)]
     pub(crate) env_store: RoverEnv,
 
     #[structopt(skip)]
     #[serde(skip_serializing)]
-    client: Client,
+    client: AtomicLazyCell<Client>,
```

Comment:
Why not just use `Option<RefCell<_>>`? If Client uses an Arc internally anyway, we can just clone the reference.

Reply:
(more like `RefCell<Option<_>>` actually)

Reply:
(more like `Cell<Option<_>>` actually)

Reply:
> Why not just use Option<RefCell<_>>? 

No reason! I just cargo culted `lazycell` since that's what cargo uses 😆 

Any chance you can elaborate on the benefits here? I guess maybe just a smaller footprint than a new dependency?

Reply:
RefCell actually, wont work with Cell...

> Any chance you can elaborate on the benefits here? I guess maybe just a smaller footprint than a new dependency?

Yes it's just to avoid extra dependencies. Trying to stick to std is good because people know how the standard library work. They don't necessarily know the other libraries. So it's a right balance to have between: how much usefulness the library is providing vs minimal dependencies.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=bc00bfb740510fb0abaf2aa85cf4bd67

> No reason! I just cargo culted `lazycell` since that's what cargo uses laughing

If you like it! :grin: It's not super important to me. You might be interested by the API of [`once_cell::sync::Lazy`](https://docs.rs/once_cell/1.7.2/once_cell/sync/struct.Lazy.html) though. @o0Ignition0o just told me you could use it in a field (you don't *have* to use it on a static, I didn't know :exploding_head:)

Reply:
and `once_cell` doesn't seem to quite fit the paradigm i'm looking for here - i'd have to write a wrapper struct around it i think